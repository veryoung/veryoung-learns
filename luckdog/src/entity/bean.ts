/** 右上角跳转链接 */
export interface RightLink {
  /** 跳转地址 */
  sUrl: string;
  /** 跳转地址文案 */
  sText: string;
}

/** 跳转更多 */
export interface MoreUrl {
  vText?: string;
}

export interface ItemBean {
  report_info?: Array<Array<string>>;
  ui_style?: number;
  parsedObject?: Record<string, any>;
  item_id?: string;
  tab_id?: string;
  title?: string;
  bottomLineStyle?: number;
  reportState?: boolean;
  symbolKey?: string;
  business?: number;
  url?: string;
}
