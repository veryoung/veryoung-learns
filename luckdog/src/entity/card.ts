/* 卡片左上角链接类型 */
export enum CardLinkType {
  NONE = 0, // 无实验下发(即本地刷新换一换，用已有的书籍进行轮换)
  MORE = 1, // 更多
  CHANGE = 2, // 局部刷新换一换（命中该实验，通过预加载远程书籍换一换）
}


/** 局部刷新换一换请求参数 */
export interface CardReportReq {
  tabId: number, // 当前tab的selectTabID
  tabIdx: number, // 卡片位于tab下的索引
  pageNum: number, // 页码
  refreshKey?: string, // 卡片的refreshKey
}


/** 局部刷新上报参数 */
export interface CardReportInfo {
  [extra: string]: string;
}
