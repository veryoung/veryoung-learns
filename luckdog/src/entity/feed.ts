import { ActiveTab, DynamicTabItem } from '@tencent/luckbox-readdata-feedsentity';

/** 本地缓存动态tab相关数据 */
export interface LocalDynamicTabs {
  dynamicTabs: DynamicTabItem[];
  activeTab: ActiveTab;
}

export * from '@tencent/luckbox-readdata-feedsentity';
