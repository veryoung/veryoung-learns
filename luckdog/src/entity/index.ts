export * from './bean';
export * from './card';
export * from './common';
export * from './feed';
export * from './skin';
export * from './tabs';
