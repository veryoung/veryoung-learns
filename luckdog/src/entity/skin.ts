/**
 * 皮肤的四种模式
 */
export enum SkinModelType {
  /** 正常 */
  NORMAL = 0,
  /** 夜间 */
  NIGHT = 1,
  /** 浅色 */
  LIGHT = 2,
  /** 深色 */
  DARK = 3,
}
