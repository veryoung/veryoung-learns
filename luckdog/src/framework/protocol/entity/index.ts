
export * from './author';
export * from './banner';
export * from './book-shelf';
export * from './books';
export * from './catch-up';
export * from './category';
export * from './kol';
export * from './nav';
export * from './tag';
export * from './video';
export * from './rank';
export * from './award';
export * from './award-record';
export * from './topic';
