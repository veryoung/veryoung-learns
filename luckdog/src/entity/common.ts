import { View } from '@tencent/hippy-react-qb';
import { ItemBean } from './bean';

export interface CommonProps {
  /** 父元素 */
  parent?: any;
  parents?: any;
  /** 全局变量 */
  globalConf?: any;

  itemBean?: ItemBean;

  index?: number;
  selectTabID?: number;

  /** 组件渲染回调 */
  onLayout?: (event: { layout: Layout }) => void;

  doBeaconByClick?: (...args) => void;

  doBeaconBySlide?: (...args) => void;
}

/** 终端布局 */
export interface Layout {
  /** 相对父元素 x轴上偏向量 */
  x: number,
  /** 相对父元素 y轴上偏向量 */
  y: number,
  /** 宽度 */
  width: number,
  /** 高度 */
  height: number
}

export type ViewRef = View;
