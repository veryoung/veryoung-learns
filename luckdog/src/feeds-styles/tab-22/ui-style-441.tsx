/**
 * UGC 话题卡片
 */
import React from 'react';
import { View, ScrollView, StyleSheet } from '@tencent/hippy-react-qb';
import { throttle } from '@/luckbox';
import { strictExposeReporter, reportUDS, BusiKey, addKeylink, logError, KeylinkCmd } from '@/luckdog';
import { CommonCardStyle } from '@/framework/FeedsConst';
import { RightBehavior, Topic, TopicCard } from '@/framework/protocol';
import { LayoutList } from '@/framework/cards/components/layout-list';
import { Title } from './components';
import FeedsUtils from '../../framework/FeedsUtils';
import { TopicItem } from './components/topic-item';

const TAG = 'ui-style-441';

/** layout-item的列/行间距 */
const ColumnSpacing = 10;
const RowSpacing = 16;

/** 话题Tag高度宽度 */
const TopicItemHeight = 20;
const TopicItemWidth = 215;

const ScrollContainerPaddingHorizontal = 14;


/** 链接跳转 */
const loadUrl = (url?: string): void => {
  if (!url) {
    logError('跳转url为空', TAG);
    return;
  }
  FeedsUtils.doLoadUrl(url);
};

/** 话题Tag曝光 */
const onTopicItemLayout = (
  event: Record<string, any>,
  cardIndex: number,
  topicId: string,
  topicIndex: number,
  layoutIdx: number,
  itemIdx: number,
): void => {
  const rect = {
    ...event.layout,
    x: itemIdx * (
      ColumnSpacing + TopicItemWidth +
      ScrollContainerPaddingHorizontal + CommonCardStyle.marginHorizontal
    ),
    y: layoutIdx * (RowSpacing + TopicItemHeight),
  };

  strictExposeReporter.addExpoItem({
    cardIndex,
    bookId: topicId,
    bookIndex: topicIndex,
    supportHorizontalScroll: true,
    rect,
  });
};

/** 更新卡片标题高度 */
const onTitleLayout = (event: Record<string, any>, cardIndex: number): void => {
  strictExposeReporter.updateTitleHeight(cardIndex, event.layout.height);
};

/** 滑动时更新Tag x坐标 */
const onTopicScroll = (event: Record<string, any>, cardIndex: number): void => {
  strictExposeReporter.updateViewportLeft(cardIndex, event.contentOffset.x);
};

interface Props {
  index: number;
  /** 当前tabid */
  selectTabID: number;
  /** 卡片后台数据 */
  itemBean: {
    parsedObject?: {
      sData: TopicCard; // 新协议结构
    }
  };
  globalConf: any;
}


export const FeedsViewUIStyle441 = (props: Props): JSX.Element | null => {
  const cardIndex = props.index;
  const { jumpLink, behavior, layoutList = [], isRow = false, title = '' } = props.itemBean.parsedObject?.sData || {};
  if (layoutList.length === 0) {
    logError(`话题卡数据异常: ${props.itemBean.parsedObject?.sData}`, TAG);
    return null;
  }
  if (layoutList.some(item => item.dataList.length !== layoutList[0].dataList.length)) {
    addKeylink(`话题卡片列数不一致: ${layoutList.map(item => item.dataList.length)}`, TAG);
    addKeylink('topic-row-columns-notmatch', KeylinkCmd.PR_ERROR_SUM);
  }

  // 根据话题index（从上往下，从左往右）将topicId重排序为一维数组
  const colNum = layoutList[0].dataList.length;
  const topIdsList: string[][] = layoutList.map(item => item.dataList.map(({ topicId }) => topicId));
  const rearrangeTopicIds: string[] = [...Array(colNum).keys()]
    .map(index => topIdsList.map(arr => arr[index]))
    .reduce((acc, item) => [...acc, ...item], []);
  strictExposeReporter.updateBookIds(cardIndex, 0, rearrangeTopicIds);

  /** 点击话题Tag */
  const onTopicItemClick = (topicUrl: string, topicId: string, topicIndex: number): void => {
    loadUrl(topicUrl);
    // 点击话题Tag上报
    reportUDS(BusiKey.CLICK__CARD, props, { ext_data1: topicIndex, ext_data2: topicId, bigdata_contentid: '' });
  };

  /** 渲染话题的一个Tag */
  const onTopicItemRender = (_isSmallItem: boolean, layoutIdx: number, itemIdx: number): JSX.Element => {
    // 计算话题index(从1开始)，从上往下，从左往右
    const topicIndex = layoutIdx + (itemIdx * layoutList.length) + 1;
    const { dataList } = layoutList[layoutIdx];
    const { topicId, topicUrl } = dataList[itemIdx];
    return <TopicItem
      topic={dataList[itemIdx]}
      height={TopicItemHeight}
      width={TopicItemWidth}
      onClick={strictExposeReporter.triggerExpoCheck(() => onTopicItemClick(topicUrl, topicId, topicIndex))}
      onLayout={event => onTopicItemLayout(event, cardIndex, topicId, topicIndex, layoutIdx, itemIdx)}
    />;
  };

  /** 点击标题右上角 */
  const onTitleRightClick = (url?: string): void => {
    // 点击查看更多上报
    if (behavior === RightBehavior.LINK_MORE) {
      loadUrl(url);
      reportUDS(BusiKey.CLICK__CARD_VIEW_ALL, props);
    }
  };

  return (
    <View
      style={CommonCardStyle}
    >
      <Title
        title={title}
        right={FeedsUtils.convertTitleRight(jumpLink)}
        rightClick={() => onTitleRightClick(jumpLink?.linkUrl)}
        onLayout={event => onTitleLayout(event, cardIndex)}
      />
      <ScrollView
        contentContainerStyle={styles.scrollContainer}
        horizontal
        showsHorizontalScrollIndicator={false}
        onScroll={throttle(event => onTopicScroll(event, cardIndex), 500, 500)}
      >
        <LayoutList<Topic>
          isRow={isRow}
          columnSpacing={ColumnSpacing}
          rowSpacing={RowSpacing}
          layoutList={layoutList}
          onItemRender={onTopicItemRender}
        />
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  scrollContainer: {
    display: 'flex',
    flexDirection: 'row',
    paddingHorizontal: ScrollContainerPaddingHorizontal,
    paddingTop: 8,
    marginBottom: 16,
  },
});
