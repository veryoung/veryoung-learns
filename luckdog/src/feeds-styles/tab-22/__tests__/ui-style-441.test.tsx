import React from 'react';
import renderer from 'react-test-renderer';
import { TabId } from '../../../entity';
import { FeedsViewUIStyle441 } from '../ui-style-441';
import { ITEM_BEAN } from './mocks/ui-style-441';

describe('<FeedsViewUIStyle441/>', () => {
  describe('render testing', () => {
    it('render style441', () => {
      const props = {
        index: 0,
        selectTabID: TabId.BOTTOM_RECOMM1,
        itemBean: {
          parsedObject: {
            sData: ITEM_BEAN,
          },
        },
        globalConf: {},
      };
      const tree = renderer.create(<FeedsViewUIStyle441 {...props} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
