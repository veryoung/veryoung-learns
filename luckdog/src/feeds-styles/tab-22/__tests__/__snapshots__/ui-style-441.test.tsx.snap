// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedsViewUIStyle441/> render testing render style441 1`] = `
<div
  nativeName="View"
  style={
    Object {
      "backgroundColors": Array [
        "white",
        "#23282d",
        "white",
        "rgba(0, 0, 0, 0.8)",
      ],
      "borderRadius": 12,
      "marginBottom": 0,
      "marginHorizontal": 12,
      "marginTop": 16,
    }
  }
>
  <div
    nativeName="View"
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
      }
    }
  >
    <p
      nativeName="Text"
      numberOfLines={1}
      style={
        Array [
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": 20,
            "maxWidth": 252,
            "paddingBottom": 14,
            "paddingTop": 14,
          },
          Object {},
        ]
      }
      text="书荒补给站"
    >
      书荒补给站
    </p>
    <div
      className="right-view"
      nativeName="View"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingBottom": 14,
          "paddingTop": 14,
          "position": "relative",
        }
      }
    >
      <p
        nativeName="Text"
        numberOfLines={1}
        style={
          Array [
            Object {
              "colors": Array [
                "#000000",
                "#686D74",
                "#000000",
                "#ffffff",
              ],
              "fontSize": 12,
              "lineHeight": 16,
            },
            Object {},
          ]
        }
        text="查看全部"
      >
        查看全部
      </p>
      <img
        alt=""
        nativeName="Image"
        noPicMode={
          Object {
            "enable": false,
          }
        }
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA8AAAAbCAYAAACjkdXHAAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAD6ADAAQAAAABAAAAGwAAAACSVupLAAAA40lEQVQ4EZ1UOQ4CMQyMQDQ8iIqWJ3HDAhXiEJf2KZS0PAX2B7QwRrFkuZo40sheecZJ1nZSSmkDNMACKF5vKL4Z2xJ1G+QWMMiiPmwXeORvysjRdXexe0plSGv4NsHBxCi3cgmOlMqQli7BycQoV8pmr3CmVIY0cwkuJka5U5fgSqkMaeIS3EyMcscuQS0q6TBmPUH6ANqJPfgdRmg5d3xoFWSY6DUEU4Vi6SEaOeH/zsy24b8drnO4w+bujnRvh6cqPM8rd1T6JQm/Yb4cO6b4ypEe1bajW07F8uS8AOmkovUDoSNiNZ1r/OEAAAAASUVORK5CYII="
        style={
          Object {
            "height": 9,
            "marginLeft": 4,
            "tintColors": Array [
              "#000000",
              "#686D74",
              "#000000",
              "#ffffff",
            ],
            "width": 5,
          }
        }
      />
    </div>
  </div>
  <div
    contentContainerStyle={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 16,
        "paddingHorizontal": 14,
        "paddingTop": 8,
      }
    }
    horizontal={true}
    nativeName="ScrollView"
    onScroll={[Function]}
    showsHorizontalScrollIndicator={false}
    style={
      Object {
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "scroll",
      }
    }
  >
    <div
      nativeName="View"
      style={
        Array [
          Object {
            "collapse": false,
            "flexDirection": "row",
          },
          Object {
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 16,
            "paddingHorizontal": 14,
            "paddingTop": 8,
          },
        ]
      }
    >
      <div
        nativeName="View"
        style={
          Object {
            "collapsable": false,
            "flexDirection": "column",
          }
        }
      >
        <div
          nativeName="View"
          style={Object {}}
        >
          <div
            nativeName="View"
            style={
              Object {
                "collapsable": false,
                "flexDirection": "row",
              }
            }
          >
            <div
              nativeName="View"
              onLayout={[Function]}
              style={Object {}}
            >
              <div
                nativeName="View"
                onClick={[Function]}
                onLayout={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "width": 215,
                  }
                }
              >
                <img
                  alt=""
                  nativeName="Image"
                  noPicMode={
                    Object {
                      "enable": false,
                    }
                  }
                  src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADwAAAA8CAMAAAANIilAAAAAIVBMVEVHcEz/cE//b0z/bkz/bkz/bk3/b07/bk3/bkz/bkz/dVWOi8B6AAAAC3RSTlMAL2aU/8FCseD2Ews4zjUAAADbSURBVHja7ZZLDsMgDAVr8zPc/8AVZvckN49IbbpgVh7JBGQw5HU4PIKootKknHNBJal5UkHpiSeKytHWVKgc5tmCytE9e6BSDE/uqBeM6qhnm8eg8qHI11hQN80MFm0QhdxddVx16ZkgRa2UirM+4iFqU+6MoO6cEUPlkLVMVA717Aa6NTiB/u4qUFSO4tkCunWPDFQO6H3QAG3FWdkeoiYJi0zQJTqU95tKMkWLikwQ7Xc1atURUie2ZvAYdfzZc4FVN9Cth72BfuEqoH8p+HZOqDSqgno4PMEb91MO0cxvjQoAAAAASUVORK5CYII="
                  style={
                    Object {
                      "height": 20,
                      "width": 20,
                    }
                  }
                />
                <p
                  nativeName="Text"
                  numberOfLines={1}
                  style={
                    Object {
                      "colors": Array [
                        "#000000",
                        "#686D74",
                        "#000000",
                        "#ffffff",
                      ],
                      "fontSize": 14,
                      "lineHeight": 20,
                      "marginLeft": 2,
                      "width": 195,
                    }
                  }
                  text="快穿也内卷，我穿到的身体多奇葩"
                >
                  快穿也内卷，我穿到的身体多奇葩
                </p>
              </div>
            </div>
            <div
              nativeName="View"
              onLayout={[Function]}
              style={
                Object {
                  "marginLeft": 10,
                }
              }
            >
              <div
                nativeName="View"
                onClick={[Function]}
                onLayout={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "width": 215,
                  }
                }
              >
                <img
                  alt=""
                  nativeName="Image"
                  noPicMode={
                    Object {
                      "enable": false,
                    }
                  }
                  src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADwAAAA8CAMAAAANIilAAAAAIVBMVEVHcEz/cE//b0z/bkz/bkz/bk3/b07/bk3/bkz/bkz/dVWOi8B6AAAAC3RSTlMAL2aU/8FCseD2Ews4zjUAAADbSURBVHja7ZZLDsMgDAVr8zPc/8AVZvckN49IbbpgVh7JBGQw5HU4PIKootKknHNBJal5UkHpiSeKytHWVKgc5tmCytE9e6BSDE/uqBeM6qhnm8eg8qHI11hQN80MFm0QhdxddVx16ZkgRa2UirM+4iFqU+6MoO6cEUPlkLVMVA717Aa6NTiB/u4qUFSO4tkCunWPDFQO6H3QAG3FWdkeoiYJi0zQJTqU95tKMkWLikwQ7Xc1atURUie2ZvAYdfzZc4FVN9Cth72BfuEqoH8p+HZOqDSqgno4PMEb91MO0cxvjQoAAAAASUVORK5CYII="
                  style={
                    Object {
                      "height": 20,
                      "width": 20,
                    }
                  }
                />
                <p
                  nativeName="Text"
                  numberOfLines={1}
                  style={
                    Object {
                      "colors": Array [
                        "#000000",
                        "#686D74",
                        "#000000",
                        "#ffffff",
                      ],
                      "fontSize": 14,
                      "lineHeight": 20,
                      "marginLeft": 2,
                      "width": 195,
                    }
                  }
                  text="最奇葩的金手指大盘点"
                >
                  最奇葩的金手指大盘点
                </p>
              </div>
            </div>
            <div
              nativeName="View"
              onLayout={[Function]}
              style={
                Object {
                  "marginLeft": 10,
                }
              }
            >
              <div
                nativeName="View"
                onClick={[Function]}
                onLayout={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "width": 215,
                  }
                }
              >
                <img
                  alt=""
                  nativeName="Image"
                  noPicMode={
                    Object {
                      "enable": false,
                    }
                  }
                  src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADwAAAA8CAMAAAANIilAAAAAIVBMVEVHcEz/cE//b0z/bkz/bkz/bk3/b07/bk3/bkz/bkz/dVWOi8B6AAAAC3RSTlMAL2aU/8FCseD2Ews4zjUAAADbSURBVHja7ZZLDsMgDAVr8zPc/8AVZvckN49IbbpgVh7JBGQw5HU4PIKootKknHNBJal5UkHpiSeKytHWVKgc5tmCytE9e6BSDE/uqBeM6qhnm8eg8qHI11hQN80MFm0QhdxddVx16ZkgRa2UirM+4iFqU+6MoO6cEUPlkLVMVA717Aa6NTiB/u4qUFSO4tkCunWPDFQO6H3QAG3FWdkeoiYJi0zQJTqU95tKMkWLikwQ7Xc1atURUie2ZvAYdfzZc4FVN9Cth72BfuEqoH8p+HZOqDSqgno4PMEb91MO0cxvjQoAAAAASUVORK5CYII="
                  style={
                    Object {
                      "height": 20,
                      "width": 20,
                    }
                  }
                />
                <p
                  nativeName="Text"
                  numberOfLines={1}
                  style={
                    Object {
                      "colors": Array [
                        "#000000",
                        "#686D74",
                        "#000000",
                        "#ffffff",
                      ],
                      "fontSize": 14,
                      "lineHeight": 20,
                      "marginLeft": 2,
                      "width": 195,
                    }
                  }
                  text="婚后甜宠文！嗑到昏迷！"
                >
                  婚后甜宠文！嗑到昏迷！
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          nativeName="View"
          style={
            Object {
              "marginTop": 16,
            }
          }
        >
          <div
            nativeName="View"
            style={
              Object {
                "collapsable": false,
                "flexDirection": "row",
              }
            }
          >
            <div
              nativeName="View"
              onLayout={[Function]}
              style={Object {}}
            >
              <div
                nativeName="View"
                onClick={[Function]}
                onLayout={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "width": 215,
                  }
                }
              >
                <img
                  alt=""
                  nativeName="Image"
                  noPicMode={
                    Object {
                      "enable": false,
                    }
                  }
                  src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADwAAAA8CAMAAAANIilAAAAAIVBMVEVHcEz/cE//b0z/bkz/bkz/bk3/b07/bk3/bkz/bkz/dVWOi8B6AAAAC3RSTlMAL2aU/8FCseD2Ews4zjUAAADbSURBVHja7ZZLDsMgDAVr8zPc/8AVZvckN49IbbpgVh7JBGQw5HU4PIKootKknHNBJal5UkHpiSeKytHWVKgc5tmCytE9e6BSDE/uqBeM6qhnm8eg8qHI11hQN80MFm0QhdxddVx16ZkgRa2UirM+4iFqU+6MoO6cEUPlkLVMVA717Aa6NTiB/u4qUFSO4tkCunWPDFQO6H3QAG3FWdkeoiYJi0zQJTqU95tKMkWLikwQ7Xc1atURUie2ZvAYdfzZc4FVN9Cth72BfuEqoH8p+HZOqDSqgno4PMEb91MO0cxvjQoAAAAASUVORK5CYII="
                  style={
                    Object {
                      "height": 20,
                      "width": 20,
                    }
                  }
                />
                <p
                  nativeName="Text"
                  numberOfLines={1}
                  style={
                    Object {
                      "colors": Array [
                        "#000000",
                        "#686D74",
                        "#000000",
                        "#ffffff",
                      ],
                      "fontSize": 14,
                      "lineHeight": 20,
                      "marginLeft": 2,
                      "width": 195,
                    }
                  }
                  text="B站崩了"
                >
                  B站崩了
                </p>
              </div>
            </div>
            <div
              nativeName="View"
              onLayout={[Function]}
              style={
                Object {
                  "marginLeft": 10,
                }
              }
            >
              <div
                nativeName="View"
                onClick={[Function]}
                onLayout={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "width": 215,
                  }
                }
              >
                <img
                  alt=""
                  nativeName="Image"
                  noPicMode={
                    Object {
                      "enable": false,
                    }
                  }
                  src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADwAAAA8CAMAAAANIilAAAAAIVBMVEVHcEz/cE//b0z/bkz/bkz/bk3/b07/bk3/bkz/bkz/dVWOi8B6AAAAC3RSTlMAL2aU/8FCseD2Ews4zjUAAADbSURBVHja7ZZLDsMgDAVr8zPc/8AVZvckN49IbbpgVh7JBGQw5HU4PIKootKknHNBJal5UkHpiSeKytHWVKgc5tmCytE9e6BSDE/uqBeM6qhnm8eg8qHI11hQN80MFm0QhdxddVx16ZkgRa2UirM+4iFqU+6MoO6cEUPlkLVMVA717Aa6NTiB/u4qUFSO4tkCunWPDFQO6H3QAG3FWdkeoiYJi0zQJTqU95tKMkWLikwQ7Xc1atURUie2ZvAYdfzZc4FVN9Cth72BfuEqoH8p+HZOqDSqgno4PMEb91MO0cxvjQoAAAAASUVORK5CYII="
                  style={
                    Object {
                      "height": 20,
                      "width": 20,
                    }
                  }
                />
                <p
                  nativeName="Text"
                  numberOfLines={1}
                  style={
                    Object {
                      "colors": Array [
                        "#000000",
                        "#686D74",
                        "#000000",
                        "#ffffff",
                      ],
                      "fontSize": 14,
                      "lineHeight": 20,
                      "marginLeft": 2,
                      "width": 195,
                    }
                  }
                  text="蒙古上单"
                >
                  蒙古上单
                </p>
              </div>
            </div>
            <div
              nativeName="View"
              onLayout={[Function]}
              style={
                Object {
                  "marginLeft": 10,
                }
              }
            >
              <div
                nativeName="View"
                onClick={[Function]}
                onLayout={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "width": 215,
                  }
                }
              >
                <img
                  alt=""
                  nativeName="Image"
                  noPicMode={
                    Object {
                      "enable": false,
                    }
                  }
                  src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADwAAAA8CAMAAAANIilAAAAAIVBMVEVHcEz/cE//b0z/bkz/bkz/bk3/b07/bk3/bkz/bkz/dVWOi8B6AAAAC3RSTlMAL2aU/8FCseD2Ews4zjUAAADbSURBVHja7ZZLDsMgDAVr8zPc/8AVZvckN49IbbpgVh7JBGQw5HU4PIKootKknHNBJal5UkHpiSeKytHWVKgc5tmCytE9e6BSDE/uqBeM6qhnm8eg8qHI11hQN80MFm0QhdxddVx16ZkgRa2UirM+4iFqU+6MoO6cEUPlkLVMVA717Aa6NTiB/u4qUFSO4tkCunWPDFQO6H3QAG3FWdkeoiYJi0zQJTqU95tKMkWLikwQ7Xc1atURUie2ZvAYdfzZc4FVN9Cth72BfuEqoH8p+HZOqDSqgno4PMEb91MO0cxvjQoAAAAASUVORK5CYII="
                  style={
                    Object {
                      "height": 20,
                      "width": 20,
                    }
                  }
                />
                <p
                  nativeName="Text"
                  numberOfLines={1}
                  style={
                    Object {
                      "colors": Array [
                        "#000000",
                        "#686D74",
                        "#000000",
                        "#ffffff",
                      ],
                      "fontSize": 14,
                      "lineHeight": 20,
                      "marginLeft": 2,
                      "width": 195,
                    }
                  }
                  text="可能发生的往事"
                >
                  可能发生的往事
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
